test/cli/dart_test.dart

```diff
@@ -24,8 +34,32 @@ void main() {
   });
 }
 
-Future<TestProcess> runSass(Iterable<String> arguments) => TestProcess.start(
-    Platform.executable,
-    ["--checked", p.absolute("bin/sass.dart")]..addAll(arguments),
-    workingDirectory: d.sandbox,
-    description: "sass");
+/// Ensures that the snapshot of the Dart executable used by [runSass] is
+/// up-to-date, if one has been generated.
+void ensureSnapshotUpToDate() {
+  for (var path in _snapshotPaths) {
+    if (new File(path).existsSync()) {
+      ensureUpToDate(path, "pub run grinder app-snapshot");
+      return;
+    }
+  }
+}
+
+Future<TestProcess> runSass(Iterable<String> arguments) {
+  var executable = _snapshotPaths.firstWhere(
+      (path) => new File(path).existsSync(),
+      orElse: () => p.absolute("bin/sass.dart"));
```

Comment:
It seems like once you build a snapshot, you're basically stuck with snapshots? (Unless you delete the build output.) Should it fall back to bin/sass.dart if the snapshots are out of date? Or if snapshots are faster, perhaps it should always require building snapshots?

Reply:
> It seems like once you build a snapshot, you're basically stuck with snapshots? (Unless you delete the build output.)

Yeah, that's right.

> Should it fall back to bin/sass.dart if the snapshots are out of date?

This would require checking if the snapshot is up-to-date for every test run, as opposed to once at the beginning of the file. I guess we could get around that with global state, but cross-test state is super risky.

> Or if snapshots are faster, perhaps it should always require building snapshots?

Snapshots are faster if you're running a whole bunch of CLI tests at once, but if you're debugging a single test it's a lot faster to just run from source. Pub has supported both modes for a long time, and I find myself using source for local debugging most of the time and using snapshots on CI and when I need to run a lot of tests at once as a sanity check.